EXPerience/contracts/EXPerienceNFT.sol

```diff
@@ -123,12 +123,12 @@ contract EXPerienceNFT is ERC721, Ownable {
 
         // Get TokenID 
         uint256 _tokenID = _totalSupply;
-        // Mint the EXPerience NFT for the address (If address already holds the NFT, _mint will revert)
-        _safeMint(msg.sender, _tokenID);
```

Comment:
When I called this state-changing internal function, I was under the impression that if for some reason it happens to fail (mostly revert but in some cases 'assert'), there might be discrepancies as tokenID was increased by mint and did not complete successfully.
I rely on the fact that require() performs a revert and this change works well.